src/NeuralPDE.jl

```diff
@@ -93,20 +93,24 @@ The PINNs Problem.
 * `train_sets`: Training sets on the domain and boundary spaces
 * `dim`: The dimensions of the problem.
 """
-struct NNPDEProblem{PDEFunction,TS} <:DiffEqBase.DEProblem
+struct NNPDEProblem{PDEFunction,Function,TS} <:DiffEqBase.DEProblem
```

Comment:
Functions probably shouldn't be held in arrays. This is going to lead to some nasty instabilities. Since the number of BC relations should be fixed, maybe a tuple makes sense? This is a detail though.

Reply:
it can be one function that returns array/tuple

Reply:
@ChrisRackauckas but I don't see a difference

Reply:
arrays of functions aren't type stable so it'll lead to type inference issues. That said, you shouldn't have tuples of more than length 16, for making it a tuple could be an issue too. This is probably something to test out before working it out.

Reply:
ok